tests/test-double-float.c

```diff
@@ -32,7 +32,8 @@ inline static float silu_float(float x) {
 int main(void) {
     uint32_t x = UINT32_MAX;
     do {
-        float f = *(float *)&x;
+        float *y = (float *)&x;
+        float f = *y;
```

Comment:
This is simply disabling the warning rather than fixing the issue. GCC is allowed to assume pointers of different types will never have the same address with -O2, so using a union or memcpy would be more appropriate, e.g.:
```patch
diff --git a/tests/test-double-float.c b/tests/test-double-float.c
index 89dafc9f..ba03f28b 100644
--- a/tests/test-double-float.c
+++ b/tests/test-double-float.c
@@ -30,9 +30,9 @@ inline static float silu_float(float x) {
 }
 
 int main(void) {
-    uint32_t x = UINT32_MAX;
+    union { uint32_t x; float f; };
+    x = UINT32_MAX;
     do {
-        float f = *(float *)&x;
         assert(!isfinite(f) || (round_orig(f) == round_float(f)));
     } while (x--);
 
```